### PR TITLE
Start writing tests

### PR DESCRIPTION
**Pull request by nadman3**
Working version with all entities and few services.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -7,6 +7,86 @@
...
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
 
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>4.2.6.RELEASE</version>
```
Comment (SvNevmerzhytska) [inline pom.xml]: you can move versions into properties in pom.xml (it is useful if you need to update version of used library in one place)

File: src/main/java/com/javaworkspace/services/CourseServiceImplementation.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import com.javaworkspace.entities.Course;
+import com.javaworkspace.repositories.CourseRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.Resource;
+import java.util.List;
+
+@Service
+public class CourseServiceImplementation implements CourseService {
```
Comment (SvNevmerzhytska) [inline src/main/java/com/javaworkspace/services/CourseServiceImplementation.java]: It is better to name classes with *Impl (not full *Implementation) (the same for other 3 classes)

File: src/main/java/com/javaworkspace/services/StudentService.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.List;
+
+public interface StudentService {
+
+    Student save(Student student);
+    void delete(Student student);
+    List findAll();
+    Student update(Student student);
+    Student findOne(Long id);
+//    void addCourseToStudent(Student student, CourseDictionary courseDictionary, Double mark);
```
Comment (SvNevmerzhytska) [inline src/main/java/com/javaworkspace/services/StudentService.java]: Why is it commented out? Fix this or remove if you aren't going to implement it.

File: src/main/java/com/javaworkspace/services/MarkServiceImplementation.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Override
+    @Transactional
+    public void delete(Mark mark) {
+        System.out.println("Deleted " + mark.getId() + " mark");
+        markRepository.delete(mark);
+    }
+
+    @Override
+    @Transactional
```
Comment (SvNevmerzhytska) [inline src/main/java/com/javaworkspace/services/MarkServiceImplementation.java]: Annotation Transactional have sense only when we make some changes in DB. That's why you can remove @Transational from all reading operations.

File: src/test/java/com/javaworkspace/TestCourseService.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {AppConfiguration.class})
+@Transactional
+public class TestCourseService {
```
Comment (SvNevmerzhytska) [inline src/test/java/com/javaworkspace/TestCourseService.java]: It is better to name test classes as CourseServiceTest. And place classes into the appropriate package under test folder (in this case com.javaworkspace.services). Package structure should be the same as the package structure of the application.

File: pom.xml (hunk 1 above)
Comment (nadman3) [inline pom.xml]: I can't find how I can make it

**Pull request merged**